apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -25,10 +27,22 @@
 
   @Nonnull ApolloCall<T> clone();
 
-  interface Callback<T> {
-    void onResponse(@Nonnull Response<T> response);
+  abstract class Callback<T> {
+    public abstract void onResponse(@Nonnull Response<T> response);
+
+    public abstract void onFailure(@Nonnull ApolloException e);
+
+    public void onHttpError(@Nonnull ApolloHttpException e) {
+      onFailure(e);
+    }
+
+    public void onNetworkError(@Nonnull ApolloNetworkException e) {
```

Comment:
I think it would be nice to pass in the status code and response here, making this look like:

```java
public void onNetworkError(int status, Response response, @Nonnull ApolloNetworkException e)
```

I fell it is nice not having to dig through the exception, secondly, it makes it clearer what we have to work with